### DISCUSSION TITLE
Need Assistance - Unrequested Child Entity Collections Being Queried

### DISCUSSION THREAD

**Discussion by CShelton11** [AutoMapper/AutoMapper.Extensions.OData#160] [General]
Sorry for the lengthy post, but trying to be detailed.

I'm experiencing an issue where information I didn't request is being included in the info returned from sql server.
I'm hoping that you might point me in the right direction here...
My setup is outlined below:

Issue By Example:
********************************************************
If I were to invoke the PolicyCase endpoint on my api with the following url: 
https://site.com/odata/policyCase?$top=1
I get a response back from the server like the following:

```
{
    "PolicyCaseId": 1,
    "TicketDetails": [
        { "TicketDetailId": 1, etc... },
        { "TicketDetailId": 2, etc... }
    ]
}
```

I didn't request the TicketDetails information in my query, but it is being provided anyway.
I can't see that I have configured anything incorrectly, and I can't believe that this isn't something that has been considered already.

Entity Classes:
********************************************************
```
public class PolicyCase {
    public Int64 PolicyCaseId { get; set; }
    public virtual ICollection<TicketDetail> TicketDetails { get; set; }
}

public class TicketDetail {
    public Int64 TicketDetailId { get; set; }
    public virtual PolicyCase PolicyCase { get; set; }
}
```

Dto Classes:
********************************************************
```
public class PolicyCaseDto {
    public Int64 PolicyCaseId { get; set; }
    public ICollection<TicketDetailDto> TicketDetails { get; set; }
}

public class TicketDetailDto {
    public Int64 TicketDetailId { get; set; }
    public PolicyCaseDto PolicyCase { get; set; }
}
```

Profiles:
********************************************************
```
public class PolicyCaseProfile : Profile {
    public PolicyCaseProfile() {
        CreateMap<PolicyCase, PolicyCaseDto>(MemberList.Destination).ReverseMap();
    }
}

public class TicketDetailProfile : Profile {
    public TicketDetailProfile() {
        CreateMap<TicketDetail, TicketDetailDto>(MemberList.Destination).ReverseMap();
    }
}
```

Mappers:
********************************************************
`_mapper = new MapperConfiguration(a => a.AddMaps(typeof(PolicyCaseProfile))).CreateMapper();`

Extension Method:
********************************************************
```
public static IQueryable<PolicyCaseDto> ToDtoQueryable(this IQueryable<PolicyCase> queryable, ODataQueryOptions<PolicyCaseDto> options) {
    return queryable.GetQuery(_mapper, options);
}

```
Note - If I were to do the following:  queryable.GetQuery(_mapper, options).ToList();
It returns all of the TicketDetails for the 1 record I filtered for.

I'm guessing it has something to do with the projection including a ToList in there somewhere, but I have to believe that this has been considered already (Like an option telling the projection to issue a Select statement instead of a ToList???).  
How do I avoid this issue????

Any help is greatly appreciated...

**Comment by CShelton11** [comment]
Well, This was entirely my fault.  I didn't look at the downstream repository.
Someone had included "Includes" statements there resulting in the unwanted data.

**Reply by CShelton11** [reply]
And it turns out, that wasn't the issue either.  
I removed Includes change tracking, and proxies from the entity queryable
Doing so caused the raw iqueryable to perform as expected.
I can also validate that the generated sql has no left join to the TicketDetails table.
When I then project that same IQueryable to a Dto IQueryable, it generates a left join to the TicketDetails table.

My guess is that the mapper is first used to project the original iqueryable to the new iqueryable.
This is resulting in an undesired inclusion of the TicketDetails and an undesired left join to that table in sql.

In my mind, one could potentially address this differently.  
Possible including an extension method with the following signature:
IMappingExpression<T, J> AddProfile<T, J>(this IMapperConfigurationExpression configuration, ODataQueryOptions options)

This will generate the mapping expression dynamically given the odata query options.
I am working on a potential solution to this right now.
Will post here when I have something workable...

**Comment by BlaiseD** [comment]
Answered in #161
